### PR TITLE
Implement concatenation operator for "u8" literals.

### PR DESCRIPTION
**Pull request by AlekseyTs**
https://github.com/dotnet/csharplang/pull/6221

Relates to test plan https://github.com/dotnet/roslyn/issues/58848

### CODE REVIEW

**Comment by AlekseyTs** [comment]
CC @davidwengier This is a small addition to the "UTF8 Literals" feature. Please let me know if you have any concerns regarding impact on IDE. Are there any specific IDE tests I should add?

**Comment by davidwengier** [comment]
Thanks, I don't think this introduces any concerns or new test areas for IDE

**Comment by AlekseyTs** [comment]
@RikkiGibson, @cston, @dotnet/roslyn-compiler Please review.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs [hunk 1]
```diff
@@ -3509,5 +3570,478 @@ class C
...
+
+    public static implicit operator ReadOnlySpan<byte>(C x) => ReadOnlySpan<byte>.Empty;
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.NetCoreApp, options: TestOptions.DebugExe);
+
+            CompileAndVerify(comp, expectedOutput: @"called", verify: Verification.Fails).VerifyDiagnostics();
+        }
+
+        [ConditionalFact(typeof(CoreClrOnly))]
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/Utf8StringsLiteralsTests.cs:3626]: >[ConditionalFact(typeof(CoreClrOnly))] [](http://example.com/codeflow?start=8&length=38)

`[Fact]`?

**Review by cston — APPROVED** [review]

**Comment by stephentoub** [comment]
Thanks!

**Pull request merged**